source/Calamari.ConsolidateCalamariPackages.Tests/IntegrationTests.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Assent;
+using FluentAssertions;
+using NSubstitute;
+using NuGet.Packaging;
+using NUnit.Framework;
+using Serilog;
+using TestStack.BDDfy;
+
+namespace Calamari.ConsolidateCalamariPackages.Tests
+{
+    [TestFixture]
+    public class IntegrationTests
+    {
+        readonly Assent.Configuration assentConfiguration = new Assent.Configuration().UsingSanitiser(s => Sanitise4PartVersions(SanitiseHashes(s)));
+        static readonly string TestPackagesDirectory = "../../../testPackages";
+
+        private string temp;
+        private string expectedZip;
+        private List<BuildPackageReference> packageReferences;
+        private bool returnValue;
+
+        public void SetUp()
+        {
+            temp = Path.GetTempFileName();
+            File.Delete(temp);
+            Directory.CreateDirectory(temp);
+            packageReferences = new List<BuildPackageReference>();
+            expectedZip = Path.Combine(temp, $"Calamari.3327050d788658cd16da010e75580d32.zip");
+        }
+
+        public void TearDown()
+        {
+            Directory.Delete(temp, true);
+            Directory.Delete(TestPackagesDirectory, true);
+        }
+
+        public void GivenABunchOfPackageReferences()
+        {
+            var artifacts = Directory.GetFiles(TestPackagesDirectory, "*.nupkg");
+            foreach (var artifact in artifacts)
+            {
+                using (var zip = ZipFile.OpenRead(artifact))
```

Comment:
I'm a little torn on whether it's worthwhile adding a Calamari.Flavour.Zip to the TestPackagesDirectory and making sure the migrated flavours handle pathing correctly. Thinking about it more it seems a bit overkill but just wanted to raise it.

Reply:
I'd like to leave it as-is for the time being, as this is the only certain input that we know of for sure, that produces what we expect. I would not be confident about modifying the expected result(s) based on the new packages we provide (unless we've thoroughly tested it in server etc which we haven't yet done).